src/App.js

```diff
@@ -155,6 +160,26 @@ class App {
 				mw.log.warn( `WhoWroteThat failed to fetch the summary for revision ${tokenInfo.revisionId}` );
 			} );
 		} );
+
+		/*
+		 * Modify fragment link scrolling behaviour to take into account the width of the infobar at
+		 * the top of the screen, to prevent the targeted heading or citation from being hidden.
+		 */
+		$( "a[href^='#']" ).on( 'click', event => {
+			var targetId, linkOffset, infobarHeight;
+			if ( !this.widget.isVisible() ) {
+				// Use the default if WWT is not active.
+				return;
+			}
+			event.preventDefault();
+			// After preventing the default event from doing it, set the URL bar fragment manually.
+			window.location.hash = targetId;
```

Comment:
This is `undefined` in my testing. Shouldn't this go below where `targedId` is set?

Reply:
Oh, you want it to *work* as well? So picky! :-P

(Fixed. And I'll write "I must not rearrange code after testing" 100 times.)